src/init/init.c

```diff
@@ -28,6 +28,16 @@
 #include <sof/schedule/ll_schedule_domain.h>
 #include <ipc/trace.h>
 
+#if CONFIG_IMX
+/*
+ * We want to avoid buggy compiler optimization (function inlining).
+ * So we replace the call to imx_addr_attr() from imx_is_cached()
+ * with a function pointer that is initialized here.
+ */
+uint32_t imx_addr_attr(void *address);
+uint32_t (*get_addr_attr)(void *address) = imx_addr_attr;
```

Comment:
This could be done in the new cache_attr file.

Reply:
Done.